### PR TITLE
Update license to EPL-2.0

### PR DESCRIPTION
**Pull request by wborn**
The license headers on all Java files have been updated to EPL-2.0.
Also updates the LICENSE file and adds a NOTICE file.

### CODE REVIEW

**Comment by mueller-ma** [comment]
Please be aware that #1356 replaces all Java files with Kotlin files and the files under https://github.com/openhab/openhab-android/tree/master/mobile/src/main/java/de/duenndns/ssl aren't under EPL-1.0.

How did you add the copyright? In a way that the copyright year can be changed easily in 2020 (#614)?

What's the motivation of this change? I've read that all Eclipse projects should migrate to EPL-2.0, but this project was never part of the Eclipse foundation.

cc @kaikreuzer

**Comment by wborn** [comment]
We've also been updating the openHAB core and add-ons project licenses from EPL-1.0 to EPL-2.0. Using a single license where possible simplifies the setup for contributors and users. The [EPL-2.0 FAQ](https://www.eclipse.org/legal/epl-2.0/faq.php) answers many questions why EPL-1.0 is deprecated and EPL-2.0 is a better license. Some highlights are:

* EPL-1.0 is more than 16 years old so EPL-2.0 uses terminology from the present time
* EPL-2.0 allows for secondary licenses
* The new headers use the SPDX-License-Identifier so tooling can easily figure out what licenses apply from analyzing the source code

I've updated these headers using the license-maven-plugin and a POM (like we do on the other projects). Obviously I didn't add the POM because this project uses Gradle. Maybe with a bit of work it can also be added as a Gradle goal.

**Comment by mueller-ma** [comment]
@wborn Can you rebase this PR and use Android Studio (AS) to apply this change? Or if you have a good automated setup just remove the current license header and I add it with AS.

**Comment by wborn** [comment]
I've updated the PR and checked how AS formats it using the following copyright template:

```
Copyright (c) 2010-$today.year Contributors to the openHAB project

See the NOTICE file(s) distributed with this work for additional
information.

This program and the accompanying materials are made available under the
terms of the Eclipse Public License 2.0 which is available at
http://www.eclipse.org/legal/epl-2.0

SPDX-License-Identifier: EPL-2.0
```

Maybe that template should be added to a file so contributors can import it? It's also stored in the files in the `.idea` dir but that is part of the `.gitignores`

**Comment by mueller-ma** [comment]
> It's also stored in the files in the .idea dir but that is part of the .gitignore

You could replace `.idea` with the recommendation of github/the community in `.gitignore` and commit the template.

**Comment by wborn** [comment]
I've updated the ignores based on the current and those in the [github/gitignore](https://github.com/github/gitignore/blob/master/Android.gitignore) project @mueller-ma. I've then added the files that were no longer ignored. AS seemed to pick them up properly after cleaning the dir and reimporting the project.

File: .gitignore [hunk 1]
```diff
@@ -1,23 +1,81 @@
...
+
+# fastlane
 fastlane/report.xml
+fastlane/Preview.html
+fastlane/screenshots
+fastlane/test_output
+fastlane/readme.md
+
+# Version control
+vcs.xml
```
Comment (mueller-ma) [inline .gitignore]: IMO vcs.xml can be included:
````
$ cat .idea/vcs.xml
<?xml version="1.0" encoding="UTF-8"?>
<project version="4">
  <component name="VcsDirectoryMappings">
    <mapping directory="$PROJECT_DIR$" vcs="Git" />
  </component>
</project>
````

Comment (wborn) [reply .gitignore]: Added!

**Comment by mueller-ma** [comment]
Looks good so far. Can you include `Fixes #614` in the commit messages?

@kaikreuzer Can you approve this?

@maniac103 We need to review if the default import order for Kotlin is fine or if we want use a custom one, like we did for Java. The import order is saved in `.idea/codeStyles/Project.xml`, which can be committed with the new `.gitignore`.

**Comment by maniac103** [comment]
@mueller-ma ktlint can generate code style files, we should probably commit those.

**Comment by mueller-ma** [comment]
@maniac103 Right, I've applied ktlint: https://github.com/mueller-ma/openhab.android/commit/20365e9e1ead637f8b4ddcdb94ac0b8e18597b5b.
@wborn Can you please cherry-pick this commit?

**Comment by wborn** [comment]
Cherry-picked!

**Comment by mueller-ma** [comment]
@wborn Can you rebase this? I'm going to merge this then.

**Comment by wborn** [comment]
I've fixed the conflicts @mueller-ma and added headers to new files where they were missing.

**Comment by BabaBabainna** [comment]
> The license headers on all Java files have been updated to EPL-2.0.
> Also updates the LICENSE

**Pull request merged**